api/v1alpha3/awscluster_types.go

```diff
@@ -89,6 +89,16 @@ type Bastion struct {
 	// with a public ip to access the VPC private network.
 	// +optional
 	Enabled bool `json:"enabled"`
+
+	// NoIngressRules will ensure no Ingress rules in the bastion host's security group.
+	// Requires CIDRBlocks to be empty.
+	// +optional
+	NoIngressRules bool `json:"noIngressRules,omitempty"`
```

Comment:
I'd probably prefer one of:
- DisableIngressRules
- DisallowIngress
- DisableIngress

@detiber I believe you previously mentioned that even if bastion.disabled is true, you thought it's still worthwhile to create the bastion security group, as users might want to BYO bastion VM, and the security group allows SSH access from the bastion's security group to the control plane & node security groups. If people are doing BYO bastion, do you think they'd be using CAPA's bastion security group for it, or something else (along with additionalSecurityGroups for the control plane and node machines)?

cc @randomvariable @rudoi @sethp-nr

Reply:
Personally think they'll be bringing their own security groups, or the bastion is more likely not even in the same VPC and is in a management one of some sort.

Reply:
Right, that's mostly what I was thinking. I would love to be able to use `Enabled` to control both the creation of the bastion instance and the security group. But that is a change in behavior, and would require manual configuration + additionalSecurityGroups to replicate.

Reply:
@ncdc I updated the boolean name to `DisableIngressRules`